src/sentry/tagstore/snuba/backend.py

```diff
@@ -706,18 +714,21 @@ def get_tag_value_paginator_for_projects(
             if converted_query is not None:
                 conditions.append([snuba_key, ">=", converted_query - FUZZY_NUMERIC_DISTANCE])
                 conditions.append([snuba_key, "<=", converted_query + FUZZY_NUMERIC_DISTANCE])
-        elif key == PROJECT_ALIAS:
+        elif include_transactions and key == PROJECT_ALIAS:
```

Comment:
Couldn't error events also have a project tag?

Reply:
So the desired behaviour here is when the user is searching in discover/performance/etc, we want to suggest Sentry project names as that's what it is expecting. But when the user is searching on the issues page, we want to suggest project tags as that's what it is expecting. So this flag is for all discover/performance/etc to turn on when we want this behaviour.

Maybe the choice of variable name is unclear 🤔 but the intent is that this is custom behaviour only for discover/performance/etc.

Reply:
Also, currently the autosuggestions for project tags on the issues page is broken because of this.

Reply:
How would users use their project tag on in discover though? Could the tag key & value autocompletion suggest `tags[project]`, and the matching values in both places? That could help educate users around how to get consistent results between issues and discover.

Reply:
Discover's search bar treats `project:xxx` as a special keyword that searches against sentry projects, not the projects tag. In order to search against their project tags, they would have to enter `tags[project]:xxx`. However, autocomplete suggestions are not supported for the `tags[xxx]:` syntax in discover.

In order to get a more consistent result across issues and discover, we would have to, like you said, suggest `tags[project]:` instead of just `project`, and include support for autocomplete suggestions for the `tags[project]:` syntax. Since neither is supported at the moment, I think this change will get us far enough until we implement them.